contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -274,21 +276,113 @@ contract WithdrawalQueue is AccessControlEnumerable, WithdrawalQueueBase {
         return _requestWithdrawalWstETH(_amountOfWstETH, _recipient);
     }
 
-    /// @notice Request withdrawal of the provided token amount in a batch
-    /// NB: Always reverts
-    function requestWithdrawalBatch(uint256[] calldata, address[] calldata)
+    struct WithdrawalRequestInput {
+        uint256 amount;
+        address recipient;
+    }
+
+    /// @notice Request the sequence of stETH withdrawals according to passed `withdrawalRequestInputs` data
+    /// @param _withdrawalRequestInputs an array of `WithdrawalRequestInput` data. The standalone withdrawal request will
+    ///  be created for each item in the passed list. If `WithdrawalRequestInput.recipient` is set to `address(0)`,
+    ///  `msg.sender` will be used as recipient.
+    /// @return requestIds an array of the created withdrawal requests
+    function requestWithdrawalBatch(WithdrawalRequestInput[] memory _withdrawalRequestInputs)
```

Comment:
Why it's not a calldata here?

Reply:
Fixed. `memory` stayed here from the previous version, where it was more gas efficient due to passing structs into the internal methods.